build/cmake/FindUtf8Proc.cmake

```diff
@@ -2,21 +2,18 @@ if(USE_BUNDLED_UTF8PROC STREQUAL "TRUE")
   message (STATUS "Enabling utf8proc (bundled)")
   add_library (utf8proc_obj OBJECT "${UTF8PROC_SOURCES}")
   set_property (TARGET utf8proc_obj PROPERTY POSITION_INDEPENDENT_CODE TRUE)
-  set (SOURCES ${SOURCES} $<TARGET_OBJECTS:utf8proc_obj>)
   target_compile_definitions (utf8proc_obj PUBLIC UTF8PROC_STATIC)
 elseif (USE_BUNDLED_UTF8PROC STREQUAL "FALSE")
  message (STATUS "Enabling utf8proc")
   find_package(PkgConfig)
   pkg_check_modules(PC_UTF8PROC REQUIRED libutf8proc)
   add_library(utf8proc STATIC IMPORTED)
-  set_target_properties(utf8proc 
-                        PROPERTIES
-                        COMPILE_OPTIONS ${PC_UTF8PROC_STATIC_CFLAGS} 
-                        LINK_LIBRARIES ${PC_UTF8PROC_LINK_LIBRARIES} 
-                        INCLUDE_DIRECTORIES ${PC_UTF8PROC_STATIC_LIBRARY_DIRS} 
-                        IMPORTED_LOCATION ${PC_UTF8PROC_LINK_LIBRARIES})
+  set_property(TARGET utf8proc PROPERTY IMPORTED_LOCATION "${PC_UTF8PROC_LINK_LIBRARIES}")
```

Comment:
On further inspection, this is incorrectly linking against the _shared_ utf8proc library, and `${PC_UTF8PROC_STATIC_LINK_LIBRARIES}` does not seem to be set when testing locally (even though `${PC_UTF8PROC_STATIC_LIBRARY_DIRS}` is set...?).

I think `find_library()` may be required to set `IMPORTED_LIBRARY` correctly:

```cmake
pkg_check_modules(PC_UTF8PROC REQUIRED libutf8proc)

find_library(
   UTF8PROC_STATIC_LIBRARY libutf8proc.a
   PATHS "${PC_UTF8PROC_STATIC_LIBRARY_DIRS}"
)

add_library(utf8proc STATIC IMPORTED)

set_property(TARGET utf8proc PROPERTY IMPORTED_LOCATION "${UTF8PROC_STATIC_LIBRARY}")
```

I think we can get away with just `libutf8proc.a` (Unix-like), but if we also want to support using system utf8proc on Windows, I think we need to add `libutf8proc.lib` to the list.

CC @vector-of-bool, mind double-checking this logic?

Reply:
That logic should work, although I would recommend `HINTS` instead of `PATHS`.

---

However, this case is mainly to satisfy requirements of system distributors, and it is likely that they would want to use the dynamic library.

I recommend using the `IMPORTED_TARGET` argument to `pkg_check_modules()` instead. It will handle generating the imported target and setting all of these properties. You can then define an `ALIAS` library to the pkg-config library, as an additional level of indirection may be required so that the rest of the build is agnostic to `USE_BUNFLED_UTF8PROC`.

This is an unfortunately subtle and non-obvious first experience with CMake 🙂.

Reply:
The way that I understood this was that @vector-of-bool's suggestion would resolve @eramongodb's concerns, so I didn't add any `find_library` calls. Let me know if that is a misunderstanding.